packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -272,6 +273,8 @@ export class TeamsBotV2Impl {
         return HostType.AppService;
       case "azure-functions":
         return HostType.Function;
+      case undefined:
+        return HostType.AppService;
```

Comment:
When will rawHostType be undefined?

Reply:
The lagacy bot. Not sure if there are better ways to handle it here.

Reply:
I guess the undefined is to handle backward compatibility, e.g., there is no host type in project settings if user created the bot by previous Teams Toolkit version, so we need to handle this case and use webapp to host it.

For this case, it is better to use [default](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in switch...case